### PR TITLE
Bug #4790 - AddBusinessDays ignores holidays when DateTime being comp…

### PR DESCRIPTION
**Pull request by adam-may**
…ared has time specified

#### Description
Addresses Issue #4790 

#### Related Issue
#4790 

#### Requires Documentation Change
None.

#### How Has This Been Tested?
Amended unit tests

#### Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [X] My code follows the code style of this project.
- [X] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [X] I have added tests to cover my changes. 
- [X] All new and existing tests passed.
- [X] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Common/Securities/Future/FuturesExpiryUtilityFunctions.cs [hunk 1]
```diff
@@ -36,21 +35,20 @@ public static class FuturesExpiryUtilityFunctions
         /// <returns>The date-time after adding n business days</returns>
         public static DateTime AddBusinessDays(DateTime time, int n, bool useEquityHolidays = true, IEnumerable<DateTime> holidayList = null)
         {
-
-            var holidays = holidayList?.ToList() ?? new List<DateTime>();
+            var holidays = holidayList?.Select(x => x.Date).ToList() ?? new List<DateTime>();
```
Comment (Martin-Molinero) [inline Common/Securities/Future/FuturesExpiryUtilityFunctions.cs:38]: A minor concern, but here we expect `holidayList` to potentially have a `time` component but the same isn't expected in the other methods `NthLastBusinessDay` & `NthBusinessDay` having the same argument, think we should standardize it to this behavior?

**Review by Martin-Molinero — COMMENTED** [review]
Nice 👍, leaving minor concern

File: Common/Securities/Future/FuturesExpiryUtilityFunctions.cs (hunk 1 above)
Comment (adam-may) [reply Common/Securities/Future/FuturesExpiryUtilityFunctions.cs:38]: @Martin-Molinero - I had wondered about that as I was making the change. Happy to update.

Comment (Martin-Molinero) [reply Common/Securities/Future/FuturesExpiryUtilityFunctions.cs:38]: Agree 👍 and avoid any deja vu of this bug

Comment (adam-may) [reply Common/Securities/Future/FuturesExpiryUtilityFunctions.cs]: @Martin-Molinero - updated with changes to other methods and tests

**Review by Martin-Molinero — APPROVED** [review]

**Pull request merged**